istioctl/cmd/wait.go

```diff
@@ -241,28 +243,14 @@ func getAndWatchResource(ictx context.Context) *watcher {
 		version := targetSchema.Resource().Version()
 		resource := collectionParts[3]
 		r := dclient.Resource(schema.GroupVersionResource{Group: group, Version: version, Resource: resource}).Namespace(namespace)
-		obj, err := r.Get(context.TODO(), nameflag, metav1.GetOptions{})
+		watch, err := r.Watch(context.TODO(), metav1.ListOptions{FieldSelector: "metadata.name=" + nf})
 		if err != nil {
 			return err
 		}
-		localResourceVersion := obj.GetResourceVersion()
-		result <- localResourceVersion
-		watch, err := r.Watch(context.TODO(), metav1.ListOptions{ResourceVersion: localResourceVersion})
-		if err != nil {
-			return err
-		}
-		metaAccessor := meta.NewAccessor()
 		for w := range watch.ResultChan() {
-			watchname, err := metaAccessor.Name(w.Object)
-			if err != nil {
-				return err
-			}
-			if watchname == nameflag {
-				newVersion, err := metaAccessor.ResourceVersion(w.Object)
-				if err != nil {
-					return err
-				}
-				result <- newVersion
+			o := w.Object.(metav1.Object)
+			if o.GetName() == nf {
+				result <- strconv.FormatInt(o.GetGeneration(), 10)
```

Comment:
don't we want observed generation, not generation? otherwise its just testing kubernetes works

Reply:
Along those lines, when testing locally I found sometimes we "miss" a generation. Should we add a test doing N writes and ensure we get N updates?

Reply:
No.  The Wait command does not make use of the status field at all, as it predates status.  Once status is stable, wait will be removed, or redirected to kubectl wait.

Reply:
Ok, in that case I don't know why, but I have been playing around with this and it does not work.
```
$ ik x wait --generation=13 VirtualService reviews.default --log_output_level=all:debug
Error: timeout expired before resource VirtualService/default/reviews became effective on all sidecars
```

But the VS is generation 13 (both in k8s and status). Not sure why